src/main/java/oracle/Parser.java

```diff
@@ -0,0 +1,120 @@
+package oracle;
+
+import command.*;
+
+public class Parser {
+    private final Ui ui;
+
+    /**
+     * Create a new Parser object
+     * @param ui stores ui given to log errors if they occur.
+     */
+    public Parser(Ui ui) {
+        this.ui = ui;
+    }
+
+    /**
+     * Parses the input gotten by the Ui, returns a command to the Oracle to execute, but doesnt execute anything
+     * by itself
+     * @param input this is the raw String given by the user
+     * @return a Command
+     */
+    public Command parse(String input) {
```

Comment:
Is this method a bit too lengthy, can consider extract out the common part of the code.